features/custom_codecs/README.md

```diff
@@ -0,0 +1,379 @@
+register new ones*.
+
+The third step is to associate your `CodecRegistry` instance with your `Cluster` instance:
+
+```java
+Cluster cluster = new Cluster.builder().withCodecRegistry(myCodecRegistry).build();
+```
+
+From now on, your codec is fully operational. It will be used every time the driver encounters
+a `MyPojo` instance when executing queries, or when you ask it to retrieve a `MyPojo` instance from a `ResultSet`.
+
+For example, here is how to save a `MyPojo` object:
+
+```java
+MyPojo myPojo = ...
+// Using SimpleStatement
+Statement stmt = new SimpleStatement("INSERT INTO t (id, json) VALUES (?, ?)", 42, myPojo));
+// Using the Query Builder
+BuiltStatement insertStmt = insertInto("t")
+    .value("id", 42)
+    .value("json", myPojo);    
+// Using BoundStatements
+PreparedStatement ps = session.prepare("INSERT INTO t (id, json) VALUES (?, ?)");
+BoundStatement bs1 = ps.bind(42, myPojo); // or alternatively...
+BoundStatement bs2 = ps.bind()
+    .setInt(0, 42)
+    .set(1, myPojo, MyPojo.class); //newly-created set() method
+```
+
+And here is how to retrieve a `MyPojo` object converted from a JSON document:
+
+```java
+ResultSet rows = session.execute(...);
+Row row = one();
+// Let the driver convert the string for you...
+MyPojo myPojo = row.get(1, MyPojo.class); // newly-created get() method
+// ... or retrieve the raw string if you need it
+String json = row.get(1, String.class); // row.getString(1) would have worked too
+```
+
+Tip: `Row` and `BoundStatement` have several methods to set and retrieve values. 
+But if you plan to use custom codecs,
+make sure that you use preferably the new `get()` and `set()` methods: these new methods 
+avoid any ambiguity by requiring the user to explicitly specify the desired Java type,
+thus forcing the driver to pick the right codec for the right task. 
+
+### Support for parameterized types
+
+The new `get()` and `set()` methods can even be used to retrieve collections or other
+user-defined parameterized types; simply use the `get()` and `set()` variants that
+take a Guava's [TypeToken][TypeToken] parameter:
+
+```java
+Row row = ...
+List<MyPojo> myPojos = row.get("jsonList", new TypeToken<List<MyPojo>>(){}); 
+// or alternatively, the following also works for collections:
+List<MyPojo> myPojos = row.getList("jsonList", MyPojo.class); 
+```
+
+### Support for collections, user types, tuples and Java Enums
+
+`CodecRegistry` instances not only store default codecs and user-defined codecs;
+they can also create new codecs on the fly, based on the set of codecs they currently hold.
+
+* Codecs for [UserType][UserType] instances are created on the fly using [UDTCodec][UDTCodec];
+* Codecs for [TupleType][TupleType] instances are created on the fly using [TupleCodec][TupleCodec];
+* Codecs for Java Enums are created on the fly using [EnumStringCodec][EnumStringCodec];
+* Codecs for collections of Enums, user types and tuple types are created on the fly using
```

Comment:
> Codecs for collections of Enums, user types and tuple types are created on the fly using
    [ListCodec][ListCodec], [SetCodec][SetCodec] and [MapCodec][MapCodec].

This is not restricted to enum, UDTs and tuples. Suggestion:

"Codecs for collections are created on the fly using [ListCodec], [SetCodec] and [MapCodec], if their element types can be handled by existing codecs (or codecs that can themselves be generated)".